docs/reference/settings.md

```diff
@@ -273,6 +273,36 @@ package = false
 
 ---
 
+### [`required-platforms`](#required-platforms) {: #required-platforms }
```

Comment:
I think it might help if the docs specifically "compared and contrasted" `required-platforms` with `environments`. They do different things of course, but they take the same inputs and I could see end users being confused by them.

(What I tend to do in docs is to mention the other in the docs and contrast them. e.g., mention `environments` in this section and `required-platforms` in the `environments` section. That way, users see the contrasting doc regardless of where they look.)